benchmarks/brute_force_vs_bvh/brute_force_vs_bvh.cpp

```diff
@@ -0,0 +1,170 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_BruteForce.hpp>
+#include <ArborX_HyperBox.hpp>
+#include <ArborX_HyperPoint.hpp>
+#include <ArborX_HyperSphere.hpp>
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <boost/program_options.hpp>
+
+template <int DIM>
+struct Dummy
+{
+  int count;
+};
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+template <int DIM>
+struct ArborX::AccessTraits<Dummy<DIM>, ArborX::PrimitivesTag>
+{
+  using memory_space = MemorySpace;
+  using size_type = typename MemorySpace::size_type;
+  static KOKKOS_FUNCTION size_type size(Dummy<DIM> const &d) { return d.count; }
+  static KOKKOS_FUNCTION auto get(Dummy<DIM> const &, size_type i)
+  {
+    ArborX::ExperimentalHyperGeometry::Point<DIM> point;
+    for (int d = 0; d < DIM; ++d)
+      point[d] = (float)i;
+    return point;
+  }
+};
+
+template <int DIM>
+struct ArborX::AccessTraits<Dummy<DIM>, ArborX::PredicatesTag>
+{
+  using memory_space = MemorySpace;
+  using size_type = typename MemorySpace::size_type;
+  static KOKKOS_FUNCTION size_type size(Dummy<DIM> const &d) { return d.count; }
+  static KOKKOS_FUNCTION auto get(Dummy<DIM> const &, size_type i)
+  {
+    ArborX::ExperimentalHyperGeometry::Point<DIM> center;
+    for (int d = 0; d < DIM; ++d)
+      center[d] = (float)i;
+    return attach(intersects(ArborX::ExperimentalHyperGeometry::Sphere<DIM>{
+                      center, (float)i}),
+                  i);
+  }
+};
```

Comment:
Can you please add a comment here describing what's happening here? I know that it's pre-existing but I think it's difficult to figure it out immediately. Just something like "The primitives and predicates are constructed on-the-fly. The ith primitive is a Point with all coordinates i. The ith predicate describes an intersection with a Sphere with center at (i,i,...) and radius i"

Reply:
Done.